packages/webdriverio/src/commands/element/click.ts

```diff
@@ -84,79 +84,93 @@ const log = logger('webdriver')
     }
+}
 
-    if (typeof options !== 'object' || Array.isArray(options)) {
-        throw new TypeError('Options must be an object')
+async function actionClick(element: WebdriverIO.Element, options: Partial<ClickOptions>) {
+    const defaultOptions: ClickOptions = {
+        button: 0,
+        x: 0,
+        y: 0,
+        skipRelease: false,
     }
 
-    let button = (options.button || 0) as Button
-    const {
-        x: xOffset = 0,
-        y: yOffset = 0,
-        skipRelease = false
-    } = options || {}
+    const { button, x, y, skipRelease }: ClickOptions = { ...defaultOptions, ...options }
 
     if (
-        typeof xOffset !== 'number'
-        || typeof yOffset !== 'number'
-        || !Number.isInteger(xOffset)
-        || !Number.isInteger(yOffset)) {
+        typeof x !== 'number'
+        || typeof y !== 'number'
+        || !Number.isInteger(x)
+        || !Number.isInteger(y)
+    ) {
         throw new TypeError('Coordinates must be integers')
     }
 
-    if (options.button === 'left') {
-        button = 0
-    }
-    if (options.button === 'middle') {
-        button = 1
-    }
-    if (options.button === 'right') {
-        button = 2
-    }
-    if (![0, 1, 2].includes(button as number)) {
+    if (!buttonValue.includes(button)) {
         throw new Error('Button type not supported.')
     }
 
-    const browser = getBrowserObject(this)
-    if (xOffset || yOffset) {
-        const { width, height } = await browser.getElementRect(this.elementId)
-        if ((xOffset && xOffset < (-Math.floor(width / 2))) || (xOffset && xOffset > Math.floor(width / 2))) {
-            log.warn('xOffset would cause a out of bounds error as it goes outside of element')
+    const browser = getBrowserObject(element)
+    if (x || y) {
+        const { width, height } = await browser.getElementRect(element.elementId)
+        if ((x && x < (-Math.floor(width / 2))) || (x && x > Math.floor(width / 2))) {
+            log.warn('x would cause a out of bounds error as it goes outside of element')
         }
-        if ((yOffset && yOffset < (-Math.floor(height / 2))) || (yOffset && yOffset > Math.floor(height / 2))) {
-            log.warn('yOffset would cause a out of bounds error as it goes outside of element')
+        if ((y && y < (-Math.floor(height / 2))) || (y && y > Math.floor(height / 2))) {
+            log.warn('y would cause a out of bounds error as it goes outside of element')
         }
     }
     const clickNested = async () => {
         await browser.action('pointer', {
             parameters: { pointerType: 'mouse' }
         })
-            .move({
-                origin: this,
-                x: xOffset,
-                y: yOffset
-            })
+            .move({ origin: element, x, y })
             .down({ button })
             .up({ button })
             .perform(skipRelease)
     }
     try {
-        await clickNested()
+        return clickNested()
```

Comment:
This would never throw because `clickNested` returns a Promise, I'ld suggest to revert this change.

Reply:
Could you explain the difference between these two when it comes to throwing please?
```js
async function example() {
  await clickNested() // can throw
}
```
async function example() {
  return clickNested() // can throw
}
```

Reply:
Both are throwing for me:
<img width="761" alt="Screenshot 2024-03-01 at 10 37 33" src="https://github.com/webdriverio/webdriverio/assets/15839059/07f4ab3c-c79f-478e-8d2a-7aabe626abab">

Reply:
So does this example:
<img width="608" alt="Screenshot 2024-03-01 at 10 40 10" src="https://github.com/webdriverio/webdriverio/assets/15839059/ee2d57f7-9cd2-4e1e-ba49-56080610c09e">